eslint-config/rules/plugin-jsdoc.js

```diff
@@ -0,0 +1,247 @@
+    'jsdoc/empty-tags': ['off'],
+
+
+    /**
+     * Reports an issue with any non-constructor function using @implements.
+     */
+    'jsdoc/implements-on-classes': ['off'],
+
+
+    /**
+     * Enforces a regular expression pattern on descriptions.
+     */
+    'jsdoc/match-description': ['off'],
+
+
+    /**
+     * Enforces a consistent padding of the block description.
+     */
+    'jsdoc/newline-after-description': ['off'],
+
+
+    /**
+     * Checks for multi-line-style comments which fail to meet the criteria of a jsdoc block.
+     */
+    'jsdoc/no-bad-blocks': ['off'],
+
+
+    /**
+     * Reports defaults being used on the relevant portion of `@param` or `@default`.
+     */
+    'jsdoc/no-defaults': ['off'],
+
+
+    /**
+     * This rule reports types being used on `@param` or `@returns`.
+     */
+    'jsdoc/no-types': ['off'],
+
+
+    /**
+     * Checks that types in jsdoc comments are defined. This can be used to check unimported types.
+     */
+    'jsdoc/no-undefined-types': ['warn'],
+
+
+    /**
+     * Requires that block description, explicit `@description`, and `@param`/`@returns` tag descriptions are written in complete sentences
+     */
+    'jsdoc/require-description-complete-sentence': ['off'],
+
+
+    /**
+     * Requires that all functions have a description.
+     */
+    'jsdoc/require-description': ['off'],
+
+
+    /**
+     * Requires that all functions have examples.
+     */
+    'jsdoc/require-example': ['off'],
+
+
+    /**
+     * Checks that all files have a `@file`, `@fileoverview`, or `@overview` tag.
+     */
+    'jsdoc/require-file-overview': ['off'],
+
+
+    /**
+     * Requires a hyphen before the `@param` description.
+     *
+     * ಠ_ಠ
+     */
+    'jsdoc/require-hyphen-before-param-description': ['off'],
+
+
+    /**
+     * Checks for presence of jsdoc comments, on class declarations as well as functions.
+     */
+    'jsdoc/require-jsdoc': ['warn', {
+      require: {
+        FunctionDeclaration: true,
+        MethodDefinition: true,
+        ClassDeclaration: true,
+        ArrowFunctionExpression: false,
+        FunctionExpression: false,
+      },
+    }],
+
+
+    /**
+     * Requires that each `@param` tag has a `description` value.
+     */
+    'jsdoc/require-param-description': ['off'],
+
+
+    /**
+     * Requires that each `@param` tag has a `name` value
+     */
+    'jsdoc/require-param-name': ['warn'],
+
+
+    /**
+     * Requires that each `@param` tag has a `type` value.
+     */
+    'jsdoc/require-param-type': ['warn'],
+
+
+    /**
+     * Requires that all function parameters are documented.
+     */
+    'jsdoc/require-param': ['warn', {
+      exemptedBy: ['endpoint'],
+    }],
+
+
+    /**
+     * Requires that each `@property` tag has a `description` value.
+     */
+    'jsdoc/require-property-description': ['off'],
+
+
+    /**
+     * Requires that each `@property` tag has a `name` value.
+     */
+    'jsdoc/require-property-name': ['off'],
```

Comment:
For consistency with `@param` rules, could we also warn on name/type for `@property`? I don't think `@property` itself should be required, but if it's there we should at least be consistent with other parameter defining tags.

Reply:
No objections to enabling this